suricata/update/rule.py

```diff
@@ -310,8 +313,10 @@ def parse_fileobj(fileobj, group=None):
         try:
             rule = parse(buf, group)
             if rule:
+                if not rule["sid"]:
+                    raise BadSidError("Sid cannot be of type null")
                 rules.append(rule)
-        except Exception as err:
+        except (Exception, BadSidError) as err:
```

Comment:
`Exception` is the base class for `BadSidError` as well, you do not need to separately add it to the except block.